qemu/tests/seabios.py

```diff
@@ -34,24 +35,104 @@ def boot_menu_check():
+    seabios_session = vm.logsessions['seabios']
+
+    if 'machine_type_bin_check' in params:
+        login_timeout = float(params.get("login_timeout", 360))
+        qemu_binary = utils_misc.get_qemu_binary(params)
+        (support_machine_types, system_versions) = utils_misc.get_support_machine_type(qemu_binary)
+
+        for m_type in support_machine_types:
+            #Using Q35 machine which is not yet fully tested on avocado-vt,so just skip it now
+            if m_type == "q35":
+                break
+            params["machine_type"] = m_type
+            params["start_vm"] = "yes"
+
+            pattern_version1 = "rhel6"
+            pattern_version2 = "rhel7"
+            pattern_version3 = "pc"
+            # pattern_version4 = "q35"
+            match_rhel6 = re.search(pattern_version1, m_type)
+            match_rhel7 = re.search(pattern_version2, m_type)
+            match_pc = re.search(pattern_version3, m_type)
+            # match_q35 = re.search(pattern_version4,m_type)
+
+            pattern = ""
+            if match_rhel6:
+                pattern += "bios.bin"
+            elif match_rhel7 or match_pc:
+                pattern += "bios-256k.bin"
+            else:
+                err_msg = "New machine type needs to check:   is %s " % m_type
+                test.error(err_msg)
+
+            env_process.preprocess_vm(test, params, env, params.get("main_vm"))
+            vm = env.get_vm(params["main_vm"])
+            session = vm.wait_for_login(timeout=login_timeout)
+            output = vm.monitor.info("roms")
+            match_bin = re.search(pattern, output)
+            if match_bin:
+                print("Check seabios bin succeeded!")
+            else:
+                err_msg = "Check seabios bin faile when machine type is %s " % m_type
+                test.error(err_msg)
+            vm.monitor.cmd("system_reset")
+        return
+
+    if 'reboottimeout' in params:
+        rtime_cfg = int(params.get("reboottimeout"))
+        rtime_sec = rtime_cfg/1000
+        time.sleep(20)
+        if rtime_sec < 60:
+            time.sleep(60)
+        else:
+            time.sleep(rtime_sec)
+
+        output = seabios_session.get_stripped_output()
+        if 0 <= rtime_cfg <= 65535:
+            pattern = "No bootable device.  Retrying in %s seconds" % rtime_sec
+            match1 = re.search(pattern, output)
+            if match1:
+                logging.info("Reboot timeout setting works!")
+            else:
+                err_msg = "Reboot-timeout tests failed,para is %s" % rtime_sec
+                test.error(err_msg)
+        elif rtime_cfg > 65535:
+            pattern = "No bootable device. Retrying in 65 seconds"
+            match1 = re.search(pattern, output)
+            if match1:
+                logging.info("Reboot timeout setting works!")
+            else:
+                err_msg = "Reboot-timeout tests failed,para is %s" % rtime_sec
+                test.error(err_msg)
```

Comment:
line 100~ line 105 are totally the same as line 1078 ~ 113, keep one is ok

Reply:
@suqinhuang  They are different scenarios and different pattern value.  If only keep one.  how should I modify it?